ios/Classes/SwiftSchibstedAccountPlugin.swift

```diff
@@ -0,0 +1,140 @@
+import Flutter
+import UIKit
+import SchibstedAccount
+
+enum Methods: String {
+    case login
+    case logout
+}
+
+enum FlutterChannels: String {
+    case callbacks = "schibsted_account/callbacks"
+    case events = "schibsted_account/events"
+}
+
+public class SwiftSchibstedAccountPlugin: NSObject, FlutterPlugin {
+    
+    var identityUI: IdentityUI?
+    var loginEventSink: FlutterEventSink?
+    var user: User?
+    var clientConfiguration: ClientConfiguration?
+    
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let channel = FlutterMethodChannel(name: FlutterChannels.callbacks.rawValue,
+                                           binaryMessenger: registrar.messenger())
+        let instance = SwiftSchibstedAccountPlugin()
+        registrar.addMethodCallDelegate(instance, channel: channel)
+        
+        
+        let eventChannel = FlutterEventChannel(name: FlutterChannels.events.rawValue,
+                                               binaryMessenger: registrar.messenger(),
+                                               codec: FlutterJSONMethodCodec.sharedInstance())
+        eventChannel.setStreamHandler(instance)
+
+        instance.loadSchibstedAccountConfig()
+    }
+
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        switch Methods(rawValue: call.method) {
+        case .login?: login()
+        case .logout?: logout()
+        default: result(nil)
+        }
+    }
+
+    private func loadSchibstedAccountConfig() {
+        guard let url = Bundle.main.url(forResource: "SchibstedAccountConfig", withExtension: "plist"),
+            let data = try? Data(contentsOf: url),
+            let config = try? PropertyListDecoder().decode(Config.self, from: data)
+            else { return }
+        
+        clientConfiguration = ClientConfiguration(
+            environment: ClientConfiguration.Environment(rawValue: config.environment) ?? .development,
+            clientID: config.clientID,
+            clientSecret: config.clientSecret,
+            appURLScheme: config.appURLScheme
+        )
+    }
+
+    private func loadLastLoggedUser() {
+        guard let clientConfiguration = clientConfiguration else { return }
+
+        let user = User.loadLast(withConfiguration: clientConfiguration);
+        switch user.state {
+        case .loggedIn:
+            onUserLoggedIn(user: user)
+        default: break
+        }
+    }
+    
+    private func login() {
+        guard let clientConfiguration = clientConfiguration else { return }
+        
+        let config = IdentityUIConfiguration(clientConfiguration: clientConfiguration)
+        self.identityUI = IdentityUI(configuration: config)
```

Comment:
I think that you like to omit `self.` unless it's necessary - here it's there to be removed 😉

Reply:
👍